qiskit/dagcircuit/dagcircuit.py

```diff
@@ -493,57 +496,59 @@ def extend_back(self, dag, edge_map=None):
         self.compose_back(dag, edge_map)
 
     def compose_back(self, input_circuit, edge_map=None):
-        """Apply the input circuit to the output of this circuit.
+        """DEPRECATED: use DAGCircuit.compose() instead.
+        """
+        warnings.warn("dag.compose_back is deprecated, please use dag.compose.",
+                      DeprecationWarning)
+        self.compose(input_circuit, edge_map)
+
+    def compose(self, other, edge_map=None, front=False):
+        """Compose the ``other`` circuit onto the output of this circuit.
 
-        The two bases must be "compatible" or an exception occurs.
         A subset of input qubits of the input circuit are mapped
         to a subset of output qubits of this circuit.
 
+        ``other`` can be narrower or of equal width to ``self``.
+
         Args:
-            input_circuit (DAGCircuit): circuit to append
-            edge_map (dict): map {Bit: Bit} from the output wires of
-                input_circuit to input wires of self. The key and value
-                can either be of type Qubit or Clbit depending on the
-                type of the node.
+            other (DAGCircuit): circuit to compose with self
+            edge_map (dict): a {Bit: Bit} map from input wires of other
+                to output wires of self (i.e. rhs->lhs).
```

Comment:
I think this is how `edge_map` worked in the `compose_back` method, but yeah i can make it lhs->rhs

Reply:
Actually i thought about this more and I think rhs-> lhs makes more sense for a dict. Because the lhs is the base you are composing onto. The base may have more qubits than the thing you put on it, so you have to say for each new thing you add, where it should go.

So for example putting a CNOT on qubits 7 & 9
```python
circuit.append(CXGate(), [7, 9])
```
is equivalent to
```
circuit.compose(CXGate(), {0: 7, 1: 9}}
```

If it helps, when we compose at the front, then the dict would be lhs->rhs. In a way, left and right are irrelevant, it's about what is the base and what is it that is being added. So I can tweak the docstring to not say "lhs" and "rhs" here.